src/core/src/pass/manager.cpp

```diff
@@ -89,89 +114,174 @@ class stopwatch {
-
-    size_t index = 0;
-    stopwatch pass_timer;
-    stopwatch overall_timer;
-    overall_timer.start();
-    bool pass_applied = false;
-    bool function_changed = false;
-    bool needs_validate = false;
-    for (auto& pass : m_pass_list) {
-        if (m_pass_config->is_disabled(pass->get_type_info())) {
-            OPENVINO_DEBUG("Pass ", pass->get_name(), " is disabled");
-            continue;
+class Profiler {
+public:
+    explicit Profiler(std::string manager_name)
+        : m_visualize("OV_ENABLE_VISUALIZE_TRACING"),
+          m_serialize("OV_ENABLE_SERIALIZE_PASS"),
+          m_profile_pass("OV_ENABLE_PROFILE_PASS"),
+          m_manager_name(std::move(manager_name)) {
+        if (m_profile_pass.is_enabled() && !m_profile_pass.is_bool()) {
+            m_file.open(m_profile_pass.get_str(), std::ios_base::app);
         }
+    }
 
-        OV_ITT_SCOPE(FIRST_INFERENCE, ov::itt::domains::ov_pass, ov::pass::perf_counters()[pass->get_type_info()]);
+    ~Profiler() {
+        if (m_file.is_open()) {
+            m_file.close();
+        }
+    }
 
-        pass_timer.start();
+    void start_timer(const std::string& name) {
+        if (m_profile_pass.is_enabled()) {
+            stopwatches[name] = stopwatch();
+            stopwatches[name].start();
 
-        if (auto matcher_pass = dynamic_pointer_cast<MatcherPass>(pass)) {
-            // This checks is to skip the graph transformation when the graph pass relies on
-            // static shape but the function state is dynamic.
-            if (matcher_pass->get_property(PassProperty::REQUIRE_STATIC_SHAPE) && func->is_dynamic()) {
-                OPENVINO_DEBUG("Pass ",
-                               pass->get_name(),
-                               " requires static shape but the "
-                               "model is dynamic. Skipping this transformation");
-                continue;
-            }
-            // GraphRewrite is a temporary container for MatcherPass to make execution
-            // on on entire ov::Model
-            pass_applied = GraphRewrite(matcher_pass).run_on_model(func);
-        } else if (auto function_pass = dynamic_pointer_cast<ModelPass>(pass)) {
-            // This checks is to skip the graph transformation when the graph pass relies on
-            // static shape but the function state is dynamic.
-            if (function_pass->get_property(PassProperty::REQUIRE_STATIC_SHAPE) && func->is_dynamic()) {
-                OPENVINO_DEBUG("Pass ",
-                               pass->get_name(),
-                               " requires static shape but the "
-                               "model is dynamic. Skipping this transformation");
-                continue;
+            bool is_pass_manager = name == m_manager_name;
+            if (is_pass_manager) {
+                std::cout << "PassManager " << m_manager_name << " started: " << std::endl;
```

Comment:
do we need this to stdout?

Reply:
should it be parsed? if yes, it should be send to file (if file path passed)

Reply:
I believe we don't need it in the file, the transformation log already contain information about pass managers